### PR TITLE
Migrate to `personator` MD API in lieu of Address Verification

### PR DESCRIPTION
**Pull request by ybur-yug**
This includes:
- update gitignore to ignore swap files
- move status codes all to YAML config
- update client to use uniform functional process method
- update property api to separate mutation concerns
- build up response processor to be a functional core
- update specs for all new/modified features
- update mocks/stubs for external requests
- bump ver

### CODE REVIEW

File: lib/melissa_data/web_smart/response_processor.rb [hunk 1]
```diff
@@ -1,39 +1,40 @@
 module MelissaData
   module WebSmart
     module ResponseProcessor
-      def process_property(response)
-        codes = codes(response)
+      require 'yaml'
+      CODE_TYPES = [:property, :address]
```
Comment (ybur-yug) [inline lib/melissa_data/web_smart/response_processor.rb:5]: In order to add another type of response processing to get these errors in the future the steps are quite simple. First, you would add the symbol for the request name here

File: lib/melissa_data/web_smart/response_processor.rb [hunk 2]
```diff
@@ -1,39 +1,40 @@
...
-          response.merge!(success: codes_for(codes, 'success'))
+          response.merge!(success: codes_for(codes, resp_type, 'success'))
         end
       end
 
-      def codes_for(codes, type)
-        codes.map { |c| method("property_#{type}_codes").call[c.to_sym] }.compact
+      def codes_for(codes, resp_type, code_type)
+        codes.map do |code|
+          YAML.load(File.read("config/#{resp_type}_#{code_type}_codes.yml")).values.first[code.to_s]
```
Comment (ybur-yug) [inline lib/melissa_data/web_smart/response_processor.rb:18]: And then you would need to make a relevant `[response_type]_[code_type]_codes.yml` file in config. This naming convention is what makes it autonomous

File: lib/melissa_data/web_smart/response_processor.rb [hunk 3]
```diff
@@ -1,39 +1,40 @@
...
 
-      def codes(response)
-        response[:result][:code].split(",")
+      def codes(response, resp_type)
+        case resp_type
+        when 'property'
+          response[:result][:code].split(",")
+        when 'address'
+          response[:results].split(",")
+        end
```
Comment (ybur-yug) [inline lib/melissa_data/web_smart/response_processor.rb:28]: And lastly you would just have to write a rule for the case that will traverse to where the error codes are stored in the hash

File: lib/melissa_data/web_smart/xml.rb [hunk 4]
```diff
@@ -1,16 +1,6 @@
...
-
-      def initialize(xml)
-        @xml_document = xml
-      end
-
-      def children?(xml_node)
-        xml_node.children.empty?
-      end
-
+    module Formatters
```
Comment (ybur-yug) [inline lib/melissa_data/web_smart/xml.rb:3]: This debatably is worth moving to its own file/namespace?

File: spec/melissa_data/web_smart/response_processor_spec.rb [hunk 5]
```diff
@@ -1,50 +1,41 @@
...
-                YS04: "Detailed information returned" }
-
-      expect(client.property_success_codes).to eq success_codes
-    end
+  let(:client)            { MelissaData::WebSmart::Client.new }
+  let(:property_response) { { result: { code: "YS01,YS03" } } }
+  let(:address_response)  { { results: "AS01" } }
+
+  it "can get error codes" do
+    property_codes = client.codes(property_response, 'property')
```
Comment (ybur-yug) [inline spec/melissa_data/web_smart/response_processor_spec.rb:9]: Need to stop mixing single/double quotes

**Comment by jonahoffline** [comment]
@ybur-yug looks good!

**Comment by ybur-yug** [comment]
feel free to merge if tests run locally fine for you @jonahoffline

**Comment by ybur-yug** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;iS5vRBkN&#x2F;55-update-melissa-data-gem-to-with-new-md-personator-service-endpoints-better-error-handling"> Update `melissa_data` gem to with new `MD Personator Service` endpoints &amp; better error handling</a>

**Comment by jonahoffline** [comment]
just ran the specs, merging! :+1:

**Pull request merged**